sklearn/utils/stats.py

```diff
@@ -1,18 +1,43 @@
 import numpy as np
 
 from .extmath import stable_cumsum
+from sklearn.utils.fixes import _take_along_axis
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
     """
     Compute the weighted ``percentile`` of ``array`` with ``sample_weight``.
+    If ``array`` is 2D, compute weighted ``percentile`` along axis=0.
     """
-    sorted_idx = np.argsort(array)
+    n_dim = array.ndim
+    if n_dim == 0:
+        return array[()]
+    # Remove trailing single-dimensional entries from the array shape
+    if n_dim == 2 and array.shape[1] == 1:
+        array.reshape((array.shape[0],))
+    sorted_idx = np.argsort(array, axis=0)
+    sorted_weights = _take_along_axis(sample_weight, sorted_idx, axis=0)
 
     # Find index of median prediction for each sample
-    weight_cdf = stable_cumsum(sample_weight[sorted_idx])
-    percentile_idx = np.searchsorted(
-        weight_cdf, (percentile / 100.) * weight_cdf[-1])
-    # in rare cases, percentile_idx equals to len(sorted_idx)
-    percentile_idx = np.clip(percentile_idx, 0, len(sorted_idx)-1)
-    return array[sorted_idx[percentile_idx]]
+    weight_cdf = stable_cumsum(sorted_weights, axis=0)
+    adjusted_percentile = (percentile / 100.) * weight_cdf[-1]
+    if n_dim == 1:
```

Comment:
You can avoid some of these conditionals by just making the array 2d above (`array = np.reshape(array, (-1, 1))` and returning `percentile[0] if ndim == 1 else percentile` below.